uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/sub-controllers/leave-request-ctrl_test.js

```diff
@@ -1031,6 +1031,40 @@
           });
         });
 
+        describe('canSubmit', function () {
+          beforeEach(function () {
+            var status = optionGroupMock.specificValue('hrleaveandabsences_leave_request_status', 'value', '3');
+            var leaveRequest = LeaveRequestInstance.init(mockData.findBy('status_id', status));
+            leaveRequest.contact_id = CRM.vars.leaveAndAbsences.contactId.toString();
```

Comment:
i really don't like the sequence of `var`s being interrupted by statements in between, it really breaks the flow of the code. Giving you two alternatives (my preference is on the first, it has one less variable declared)
```js
beforeEach(function () {
  var status = optionGroupMock.specificValue('hrleaveandabsences_leave_request_status', 'value', '3');
  var leaveRequest = LeaveRequestInstance.init(mockData.findBy('status_id', status));

  leaveRequest.contact_id = CRM.vars.leaveAndAbsences.contactId.toString();

  initTestController({
    contactId: leaveRequest.contact_id,
    leaveRequest: leaveRequest,
    userRole: 'staff'
  });
});
```

```js
beforeEach(function () {
  var status = optionGroupMock.specificValue('hrleaveandabsences_leave_request_status', 'value', '3');
  var leaveRequest = LeaveRequestInstance.init(mockData.findBy('status_id', status));
  var directiveOptions = {
    contactId: CRM.vars.leaveAndAbsences.contactId.toString(), // staff's contact id
    leaveRequest: leaveRequest,
    userRole: 'staff'
  };

  leaveRequest.contact_id = directiveOptions.contactId;

  initTestController(directiveOptions);
});
```